onnxruntime/core/optimizer/optimizer_execution_frame.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/graph/graph.h"
+#include "core/providers/cpu/cpu_execution_provider.h"
+#include "core/framework/execution_frame.h"
+#include "core/framework/mlvalue_name_idx_map.h"
+#include "core/framework/ml_value.h"
+
+namespace onnxruntime {
+
+class OptimizerExecutionFrame final : public IExecutionFrame {
```

Comment:
The PR looks good, thanks for the work @weixingzhang.
I have two high-level comments:
- Does the CPU execution provider have to be hardcoded? I was thinking that we could eventually use other providers in this execution frame.
- Would this streamlined version of the execution frame be useful outside the optimizer? If so, we could give it a more generic name than OptimizerExecutionFrame.

Reply:
Thanks for the feedback.

The reason to use CPU execution provider is that only CPU execution provider supports all of operations with all possible inputs format. I was thinking to define Info ctor with an parameter of an instance of execution provider, that means the user of Info needs to create execution provider, which I think it is not necessary for now. Instead, it is better to centralizer the creation of execution provider.

Regarding to OptimizerExecutionFrame, I would like not to make it generic until it is really useful for other cases. Also, at that time, probably the code related to ExecutionFrame needs to be refactored again.